### PR TITLE
[MBL-1478][MBL-1482] Add Primary Action button actions for Take Survey and Confirm Address

### PR DESCRIPTION
**Pull request by mtgriego**
# 📲 What

Added the primary action button actions for take survey and confirm address PPO cards
Both should open the webview and should refresh when you leave sad webview to return to the PPO Page

# 🛠 How

Adjusted some code to make all Primary Button Actions done on the activity level}
Adjusted the result launcher and finish code to allow the PPO Activity to know to refresh

# 👀 See

https://github.com/user-attachments/assets/d8c55425-fa61-4183-9cf5-52474562741c

# 📋 QA

You should be able to click the "Take Survey" or "Edit" (Address) buttons to be taken to the Webview.  From there whether or not you made any changes when you leave the webview it should refresh the PPO list and either remove cards if the action was successful or display them still if action is still needed.

# Story 📖

[MBL-1478](https://kickstarter.atlassian.net/browse/MBL-1478)
[MBL-1482](https://kickstarter.atlassian.net/browse/MBL-1482)

[MBL-1478]: https://kickstarter.atlassian.net/browse/MBL-1478?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[MBL-1482]: https://kickstarter.atlassian.net/browse/MBL-1482?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/src/main/java/com/kickstarter/features/pledgedprojectsoverview/ui/PledgedProjectsOverviewActivity.kt [hunk 1]
```diff
@@ -86,29 +93,72 @@ class PledgedProjectsOverviewActivity : AppCompatActivity() {
...
+                        },
                         isLoading = isLoading,
                         isErrored = isErrored,
                         showEmptyState = showEmptyState,
                         onSeeAllBackedProjectsClick = { startProfileActivity() },
                         pullRefreshCallback = {
-                            // TODO call viewmodel.getPledgedProjects() here
+                            viewModel.getPledgedProjects()
+                        },
+                        onPrimaryActionButtonClicked = { PPOCard ->
```
Comment (leighdouglas) [inline app/src/main/java/com/kickstarter/features/pledgedprojectsoverview/ui/PledgedProjectsOverviewActivity.kt:114]: 👏🏽👏🏽👏🏽👏🏽

File: app/src/main/java/com/kickstarter/features/pledgedprojectsoverview/ui/PledgedProjectsOverviewActivity.kt [hunk 2]
```diff
@@ -41,16 +41,21 @@ class PledgedProjectsOverviewActivity : AppCompatActivity() {
...
+    private var startForResult =
+        registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
+            if (result.resultCode == Activity.RESULT_OK) {
+                val data = result.data?.getBooleanExtra(IntentKey.FIX_PAYMENT_SUCCESS, false)
+                data?.let {
+                    if (it.isTrue()) {
+                        viewModel.getPledgedProjects()
+                    }
+                }
+                val refresh = result.data?.getStringExtra(IntentKey.REFRESH_PPO_LIST)
```
[the remark this answers is not in the document]
Comment (mtgriego) [reply app/src/main/java/com/kickstarter/features/pledgedprojectsoverview/ui/PledgedProjectsOverviewActivity.kt:53]: I'm good with that!  I wasnt sure if we wanted to keep them seperated or not so i did for safety sake at first

[the remark this answers is not in the document]
Comment (leighdouglas) [reply app/src/main/java/com/kickstarter/features/pledgedprojectsoverview/ui/PledgedProjectsOverviewActivity.kt:53]: okay, let's merge this PR and then i'll add that to mine. there's gonna be merge conflicts anyways lol

**Review by leighdouglas — APPROVED** [review]
lgtm! 🎉

**Pull request merged**